src/patterns/api-lambda.ts

```diff
@@ -0,0 +1,31 @@
+import type { LambdaRestApiProps } from "@aws-cdk/aws-apigateway";
+import { LambdaRestApi } from "@aws-cdk/aws-apigateway";
+import type { GuLambdaErrorPercentageMonitoringProps, NoMonitoring } from "../constructs/cloudwatch";
+import type { GuStack } from "../constructs/core";
+import type { GuFunctionProps } from "../constructs/lambda";
+import { GuLambdaFunction } from "../constructs/lambda";
+
+interface ApiProps extends Omit<LambdaRestApiProps, "handler"> {
+  id: string;
+}
+
+interface GuApiLambdaProps extends Omit<GuFunctionProps, "errorPercentageMonitoring"> {
+  apis: ApiProps[];
+  monitoringConfiguration: NoMonitoring | GuLambdaErrorPercentageMonitoringProps;
+}
+
+export class GuApiLambda extends GuLambdaFunction {
```

Comment:
It's probably worth adding docs for this new pattern? Again, happy for that to follow in a separate PR if you'd prefer to get this merged.